[CDU] Improve LAT REV page layout

beheh opened the pull request:
<!-- ⚠⚠ Do not delete this pull request template! ⚠⚠ -->
<!-- Pull requests that do not follow this template are likely to be ignored. -->

## Summary of Changes
<!-- Please provide a summary of changes for this pull request, ensuring all changes are explained. -->

This PR greatly improves the look of lateral revision page (LAT REV) in the MCDU, which right now is quite a bit away from the reference documentation.

Please note that this is mostly fixes visual issues on the LAT REV page only, not any behaviours and not any of the subpages, e.g. HOLD or AIRWAYS are out of scope for this PR. Items that were non-functional are also marked in gray now, but the behaviours were never there in the first place (nothing has been removed as part of this PR).

Here's a list of all the changes:
* [x] Fix page title according to reference (colors, text size, show runway if set)
* [x] Fix coordinate format (N12.123.1 E12.123.12 => 12.123.1N/012.123.123E)
* [x] Fix position of ARRIVAL (1L => 1R)
* [x] Fix typo (WING => XING)
* [x] Show or hide items based on type of waypoint and in some cases flight phase
  * Three primary versions: departure airport, enroute waypoint, destination airport
  * ENABLE ALTN is only visible if an alternate is set on the init page, and only on enroute waypoint and destination
* [x] Change text color for items missing implementation to inop gray
* [x] Disable NEXT WPT on destination airport for now as that crashes the sim

## Screenshots (if necessary)
<!-- If your PR includes visual changes, screenshots from before and after your change should always be included. -->
<!-- Please make best efforts to provide useful before and after screenshots. They should match camera angle, zoom, size, time of day, etc. -->

![before/after](https://user-images.githubusercontent.com/929769/98873622-09dc3000-2479-11eb-88c3-e2240b62527d.png)

(note that the wrong N/E vs N/W coordinates were fixed after these screenshots were taken as part of #1867).

## References
<!-- You should be making changes based on some kind reference (manuals, videos, IRL photos). P3D/xplane/fsx references will only be accepted if we believe that one cannot reasonably obtain a better source. Please post screenshots of the references you used. Ask around in the discord for how to find references for what you are working on. Exceptions will probably be made for IRL A320 pilots and engineers. -->

![grafik](https://user-images.githubusercontent.com/929769/98860633-46516100-2464-11eb-81c9-1a4cf28c6bbb.png)


<!-- If you are making a pull request related to the MCDU, please make sure you are ONLY referencing the Honeywell Pegasus Step 1A (Rev 0), 2009 edition manual. -->
<!-- If you do not have this manual, please ask on our discord for assistance -->

## Additional context
<!-- Add any other context about the pull request here. -->

The visual layout of the page should be pretty obvious and changes depending on the type of waypoint you select (departure airport/enroute waypoint/destination waypoint). However:
- `NEXT WPT`: This will be hidden on the origin airport page as per the documentation after PREFLIGHT has ended, so you may not see it if you're in the takeoff phase.
- `NEXT WPT`: Is disabled on the destination airport for now as the sim can't handle that yet and crashes
- `HOLD`: While a hold page exists, it doesn't work really yet. I'm keeping it available for now.

The subpages (HOLD/ARRIVAL/DEPARTURE/AIRWAYS) don't cleanly return to this page yet on the return button, but I'll keep that out of scope for now as it's a bit tricky.

<!-- You may optionally provide your discord username, so that we may contact you directly about the issue. -->
Discord username (if different from GitHub): BehEh#1234

<!-- DO NOT DELETE THIS -->
## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

beheh commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_LateralRevisionPage.js:
```diff
@@ -11,35 +11,95 @@ class CDULateralRevisionPage {
         console.log(waypoint);
         if (waypoint && waypoint.infos && waypoint.infos.coordinates) {
-            coordinates = waypoint.infos.coordinates.toDegreeString();
+            const lat = CDUInitPage.ConvertDDToDMS(waypoint.infos.coordinates['lat'], false);
```
I'm not super happy about pulling this from the Init page - both the Init and the InitIRS page use this so far. Any suggestion for a more centralized place where this could live in future?

lousybyte replied:
`A320_Neo_CDU_MainDisplay.js`? Should be able to call it anywhere after with `mcdu.ConvertDDToDMS()`.

beheh replied:
Hm, it can be static though, and that might be more convenient for other places. Some kind of static helper class might be more useful. I'm just leaving this where it is now and we can pull it later if we need it in even more places.

Benjozork commented:
Hello, please rework to fit #1830

beheh commented:
Thanks, I've incorporated the new formatting helpers. Should be ready to go out of draft soon.

beheh commented:
Note to self: Double-check pseudo waypoints (DECEL)

danbw9 commented:
Name: danbw9#7028
Date of testing: 12/11/2020
Version of the sim: 1.10.11.0
PR Tested: #1845
Object of test: Test new visual changes to the LAT REV page

Testing technique: Identify the issues with the current version of LAT REV page without PR with screenshots and then compare to with PR installed - and check items in PR are all correct and have no negative effect on F-PLAN page/MCDU functionality. 

Loaded up C&D in ESGG. Normal flight from ESGG to EKCH. 

In preflight stage MCDU setup:
- checked for the various types of LAT REV pages (DEP, ARR, different types of ENR waypoints) and the distinct differences that make sense - specifically the hidden and shown features when appropriate.
- tried ENABLE ALT without ALT set in INIT A and not possible. With ALT set, it's there. 
- Typo fixed now correctly showing XING

Can confirm that compared to the reference 'AFTER screenshots' - that with the coordinates PR, the text and coordinates format for waypoints make sense. 

After takeoff in CLB and CRZ was able to make lateral revisions (the ones possible) with no issues 
- direct to en-route
- change of STAR and Approach


Issues (If Any): None
Overall Rating: Good
Conclusions of the testing: LGTM. Looks great even if it's just visual changes. I wondering if all the other core devs are happy with the grey text being there when it doesn't look realistic but indicates no functionality...someone else could share their opinions on that.  Thanks for this!

wpine215 commented:
LGTM, but @beheh you need to update your fork with the current master - I can't seem to do it from the Github interface because of the force pushes that were made.

beheh commented:
I don't think it's the force pushes, it's probably this setting:

![grafik](https://user-images.githubusercontent.com/929769/99040915-464b8100-258a-11eb-8eba-d0518cd34f8a.png)

I've rebased onto master and also changed that now, so you should be able to edit. Sorry for the inconvenience.
